src/VisualStudio/Core/Def/LanguageService/AbstractCreateServicesOnTextViewConnection.cs

```diff
@@ -45,56 +47,54 @@ protected virtual Task InitializeServiceForOpenedDocumentAsync(Document document
         {
             Workspace = workspace;
             GlobalOptions = globalOptions;
-
-            _listener = listenerProvider.GetListener(FeatureAttribute.Workspace);
-            _threadingContext = threadingContext;
             _languageName = languageName;
 
-            Workspace.DocumentOpened += InitializeServiceOnDocumentOpened;
+            _workQueue = new AsyncBatchingWorkQueue<DocumentId?>(
+                    TimeSpan.FromSeconds(1),
+                    ProcessBatchDocumentOpenedAsync,
+                    EqualityComparer<DocumentId?>.Default,
+                    listenerProvider.GetListener(FeatureAttribute.Workspace),
```

Comment:
Should this be FeatureAttribute.Completion since everything here I think feeds into completion?

Reply:
I'm fine wither way, used Workspace since this component is not limited to completion even though it's only used by it at the moment

Reply:
We should just move it to completion, but we can take that as a separate change. Maybe TODO this for now?

Reply:
Fixed in this PR